src/main/java/com/microsoft/jenkins/appservice/BaseDeploymentRecorder.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.jenkins.appservice;
+
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.appservice.implementation.SiteInner;
+import com.microsoft.azure.management.appservice.implementation.WebAppsInner;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+import com.microsoft.azure.util.AzureCredentials;
+import com.microsoft.jenkins.appservice.util.TokenCache;
+import hudson.Util;
+import hudson.model.AbstractProject;
+import hudson.model.Item;
+import hudson.security.ACL;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Publisher;
+import hudson.tasks.Recorder;
+import hudson.util.ListBoxModel;
+import jenkins.tasks.SimpleBuildStep;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.CheckForNull;
+import java.util.Collections;
+
+public abstract class BaseDeploymentRecorder extends Recorder implements SimpleBuildStep {
+
+    protected final String azureCredentialsId;
+    protected final String resourceGroup;
+    protected final String appName;
+
+    @CheckForNull protected String filePath;
+    @CheckForNull protected String sourceDirectory;
+    @CheckForNull protected String targetDirectory;
+    protected boolean deployOnlyIfSuccessful;
+
+    protected BaseDeploymentRecorder(String azureCredentialsId, String resourceGroup, String appName) {
```

Comment:
OK, I'll update the rule set.

@zackliu  Could you update the checkstyle configuration in azure-vm-agents plugin to make our rules consistent among all projects?